cmd/cue/cmd/version.go

```diff
@@ -49,13 +53,69 @@ var (
 
 func runVersion(cmd *Command, args []string) error {
 	w := cmd.OutOrStdout()
-	if bi, ok := debug.ReadBuildInfo(); ok && version == defaultVersion {
-		// No specific version provided via version
-		version = bi.Main.Version
+	// read in build info
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		// shouldn't happen
+		return errors.New("unknown error reading build-info")
+	}
+	mod := &bi.Main
+	if mod.Replace != nil {
+		mod = mod.Replace
+	}
+
+	// test-based overrides
+	if v := os.Getenv("CUE_VERSION_TEST_CFG"); v != "" {
+		var extra []debug.BuildSetting
+		if err := json.Unmarshal([]byte(v), &extra); err != nil {
+			return err
+		}
+		bi.Settings = append(bi.Settings, extra...)
+	}
+
+	// prefer ldflags `version` override
+	if version == defaultVersion {
+		// no version provided via ldflags, try buildinfo
+		if bi.Main.Version != "" && bi.Main.Version != defaultVersion {
+			version = bi.Main.Version
+		}
+	}
+
+	if version == defaultVersion {
+		// a specific version was not provided by ldflags or buildInfo
+		// attempt to make our own
+		var vcsTime time.Time
+		var vcsRevision string
+		for _, s := range bi.Settings {
+			switch s.Key {
+			case "vcs.time":
+				// If the format is invalid, we'll print a zero timestamp.
+				vcsTime, _ = time.Parse(time.RFC3339Nano, s.Value)
+			case "vcs.revision":
+				vcsRevision = s.Value
+				if len(vcsRevision) > 12 {
+					vcsRevision = vcsRevision[:12]
+				}
+			}
+		}
+		if vcsRevision != "" {
+			version = module.PseudoVersion("", "", vcsTime, vcsRevision)
+		}
+	}
+
+	fmt.Fprintf(w, "cue version %v\n", version)
```

Comment:
```suggestion
	fmt.Fprintf(w, "cue version %v\n\n", version)
```

an extra newline helps separate the build settings in my opinion, otherwise the output is a bit cluttered